thavalon-webapp/src/components/gameContainer.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from "react";
+import { Lobby } from "./gameComponents/lobby";
+import { GameSocket, OutboundMessageType, InboundMessage, InboundMessageType } from "../utils/GameSocket";
+
+/**
+ * Props interface for the GameContainer. This structure matches the object
+ * passed by <Route /> to extract the required state information.
+ */
+interface GameContainerProps {
+    location: {
+        state: {
+            socketUrl: string,
+            friendCode: string
+        }
+    }
+}
+
+/**
+ * Enum of lobby states to determine what is shown to the user.
+ */
+enum LobbyState {
+    Lobby = "Lobby",
+    Game = "Game",
+}
+
+/**
+ * Interface for the incoming lobby state message.
+ */
+interface LobbyStateResponse {
+    state: LobbyState
+}
+
+/**
+ * Contains all of the game related components and sets up low level connection
+ * infrastructure.
+ * @param props Props object for the GameContainer
+ */
+export function GameContainer(props: GameContainerProps): JSX.Element {
+    // State to maintain a connection instance so we don't have to getInstance every time.
+    const [connection, setConnection] = useState<GameSocket | undefined>(undefined);
```

Comment:
up to you, but I think it's clearer to use `getInstance` since the `GameSocket` isn't really state owned by the `GameContainer`. Then we also don't have to worry about keeping it in sync if the socket changes